src/query/api/v1/handler/prometheus/native/complete_tags.go

```diff
@@ -86,6 +86,12 @@ func (h *CompleteTagsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
-	// TODO: Support multiple result types
-	prometheus.RenderTagCompletionResultsJSON(w, result)
+	if !result.Exhaustive {
+		w.Header().Set(handler.LimitHeader, "true")
+	}
+
+	if err = prometheus.RenderTagCompletionResultsJSON(w, result); err != nil {
+		logger.Error("unable to render results", zap.Error(err))
+		xhttp.Error(w, err, http.StatusBadRequest)
```

Comment:
Hm this is probably a 500 yeah?  Also I don't think we can write to the HTTP response if we already a partial output? (the headers and some of the body might have already been sent).

Probably best we can do is log the error.

Reply:
Is there any way we can detect failed writes? We had an issue a while ago where we had a legitimate error in the render code, but it would always return empty results